### PR TITLE
Relax keyword argument type in REPL.setup_interface()

### PR DESCRIPTION
**Pull request by carlobaldassi**
[This type annotation](https://github.com/JuliaLang/julia/blob/ee4c5bcf5864de5aad4e44222cb2990dd8d48dd7/base/repl/REPL.jl#L696), introduced in b3e658922d1f8dfbd518673bea3b1d1733d15820, does not allow to pass a `Dict` as the keyword argument `extra_repl_keymap` in the function `REPL.setup_interface()` any more.

However, [there is still code](https://github.com/JuliaLang/julia/blob/ee4c5bcf5864de5aad4e44222cb2990dd8d48dd7/base/repl/REPL.jl#L795-L798) intended for allowing to pass a `Dict`, and [the manual](https://github.com/JuliaLang/julia/blob/ee4c5bcf5864de5aad4e44222cb2990dd8d48dd7/doc/src/manual/interacting-with-julia.md#customizing-keybindings) uses it too (which is nicer from the UI perspective IMO).

I suppose this was accidental, and this PR relaxes again the type of that argument. The other option would be to remove some dead code and change the manual.

### CODE REVIEW

**Comment by carlobaldassi** [comment]
Appveyor failure is unrelated (it's in libgit2-online).

**Review by KristofferC — APPROVED** [review]

**Comment by tkelman** [comment]
can we test this?

**Comment by carlobaldassi** [comment]
Added tests (it's basically testing my REPL setup...) and rebased.

File: test/repl.jl [hunk 1]
```diff
@@ -665,3 +665,106 @@ let term = Base.Terminals.TTYTerminal("dumb",IOBuffer("1+2\n"),IOBuffer(),IOBuff
...
+
+if !is_windows() || Sys.windows_version() >= Sys.WINDOWS_VISTA_VER
+    for keys = [altkeys, merge(altkeys...)]
+        histfile = tempname()
+        try
+            stdin_write, stdout_read, stderr_read, repl = fake_repl()
+
+            repl.specialdisplay = Base.REPL.REPLDisplay(repl)
+            repl.history_file = true
+            ENV["JULIA_HISTORY"] = histfile
```
Comment (tkelman) [inline test/repl.jl]: should probably use withenv to put this back to what it was when done

Comment (carlobaldassi) [inline test/repl.jl]: ah right, will fix.

**Comment by carlobaldassi** [comment]
Addressed comments, fixed an issue on Windows, tests pass now (except for the unrelated libgit2 failure on Windows).
Good to go?

**Review by tkelman — APPROVED** [review]
lgtm

**Pull request merged**